packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -216,9 +216,10 @@ class _CupertinoSwitchState extends State<CupertinoSwitch> with TickerProviderSt
   }
 
   void _handleTapDown(TapDownDetails details) {
-    if (isInteractive)
+    if (isInteractive) {
       needsPositionAnimation = false;
       _reactionController.forward();
+    }
```

Comment:
The bug was here

Reply:
Can you add a test that verifies this? Maybe this should be separated into another change.

Reply:
After a quick look at this code that I'm quite foreign to, I don't really know how to test it :-/

Reply:
I can do it in a separate change if you'd like to revert. :) I just filed https://github.com/flutter/flutter/issues/84390 to track.

Reply:
Thanks! I just reverted the change in `cupertino/switch.dart` and rebased on master.